agent/cache/cache.go

```diff
@@ -554,9 +554,16 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 		return ch
 	}
 
-	// If we already have an entry and it is actively fetching, then return
-	// the currently active waiter.
+	// If we already have an entry, it is actively fetching, and we don't need
+	// to revalidate the cache, then return the currently active waiter.
 	if ok && entry.Fetching {
+		// If we need to revalidate, return with a closed channel
+		// so the caller will immediately retry the cache
+		if r.Info.MustRevalidate {
+			ch := make(chan struct{})
+			close(ch)
+			return ch
+		}
```

Comment:
something that I am still unsure about with this is how it affects other cache types 🤔